From a020dc7f60df1d7cb287fa95af7599b58c665c51 Mon Sep 17 00:00:00 2001
From: Quentin Smith <quentin@mit.edu>
Date: Sun, 9 Aug 2020 05:37:48 -0400
Subject: [PATCH 6/8] Fix buffer overflow in zstat

---
 clients/zstat/zstat.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/clients/zstat/zstat.c b/clients/zstat/zstat.c
index 68568938..8ff5ab89 100644
--- a/clients/zstat/zstat.c
+++ b/clients/zstat/zstat.c
@@ -163,7 +163,7 @@ hm_stat(char *host,
 	}
 
 	mp = notice.z_message;
-	for (nf=0;mp<notice.z_message+notice.z_message_len;nf++) {
+	for (nf=0;mp<notice.z_message+notice.z_message_len && nf<HM_SIZE;nf++) {
 		line[nf] = mp;
 		mp += strlen(mp)+1;
 	}
@@ -195,7 +195,7 @@ hm_stat(char *host,
 int
 srv_stat(char *host)
 {
-	char *line[20],*mp;
+	char *line[HM_SIZE],*mp;
 	int sock,i,nf,ret;
 	struct hostent *hp;
 	struct sockaddr_in sin;
-- 
2.28.0

