Index: configure.in
===================================================================
--- configure.in	(revision 3945)
+++ configure.in	(working copy)
@@ -142,6 +142,12 @@
 			       $LIBPATH -ldes)],
 		 $LIBPATH -ldes425 -lkrb5 -l${crypto} -lcom_err -lresolv)
 fi
+if test "$KRB4_LIBS" != "" ; then
+    AC_CANONICAL_TARGET
+    case "$target_os" in
+    darwin*) KRB4_LIBS="$KRB4_LIBS -framework Kerberos"
+    esac
+fi
 
 AC_MSG_CHECKING(for Kerberos 5)
 AC_ARG_WITH(krb5,
@@ -163,6 +169,12 @@
     REG_SVR_DEFS=-DKRB4
     REG_SVR_LIBS=-lkadm
 fi
+if test "$KRB5_LIBS" != "" ; then
+    AC_CANONICAL_TARGET
+    case "$target_os" in
+    darwin*) KRB5_LIBS="$KRB5_LIBS -framework Kerberos"
+    esac
+fi
 AC_SUBST(REG_SVR_DEFS)
 AC_SUBST(REG_SVR_INCLUDES)
 AC_SUBST(REG_SVR_LIBS)
Index: clients/mrtest/mrtest.c
===================================================================
--- clients/mrtest/mrtest.c	(revision 3945)
+++ clients/mrtest/mrtest.c	(working copy)
@@ -97,8 +97,13 @@
 	}
     }
 
+#if defined(__APPLE__) && defined(__MACH__)
+  add_error_table(&et_sms_error_table);
+  add_error_table(&et_krb_error_table);
+#else
   initialize_sms_error_table();
   initialize_krb_error_table();
+#endif
 
 #ifdef HAVE_READLINE
   /* we don't want filename completion */
Index: lib/mr_init.c
===================================================================
--- lib/mr_init.c	(revision 3945)
+++ lib/mr_init.c	(working copy)
@@ -20,8 +20,15 @@
   if (mr_inited)
     return;
 
+#if defined(__APPLE__) && defined(__MACH__)
+  add_error_table(&et_sms_error_table);
+  add_error_table(&et_krb_error_table);
+  add_error_table(&et_ureg_error_table);
+#else
   initialize_sms_error_table();
   initialize_krb_error_table();
   initialize_ureg_error_table();
+#endif
+
   mr_inited = 1;
 }
Index: include/Makefile.in
===================================================================
--- include/Makefile.in	(revision 3945)
+++ include/Makefile.in	(working copy)
@@ -7,6 +7,10 @@
 SRCTOP=@top_srcdir@
 BUILDTOP=../..
 
+prefix=@prefix@
+includedir=@includedir@
+INSTALL=@INSTALL@
+
 all:
 
 clean:
@@ -17,3 +21,5 @@
 depend:
 
 install:
+	$(SRCTOP)/mkinstalldirs $(DESTDIR)$(includedir)/moira
+	$(INSTALL) moira.h $(DESTDIR)$(includedir)/moira
Index: lib/Makefile.in
===================================================================
--- lib/Makefile.in	(revision 3945)
+++ lib/Makefile.in	(working copy)
@@ -25,6 +25,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
+includedir=@includedir@
 
 OBJS=	critical.lo fixhost.lo fixname.lo \
 	hash.lo kname_unparse.lo kname_parse.lo krb_et.lo mr_access.lo mr_auth.lo \
@@ -54,6 +55,8 @@
 
 install: all
 	$(LIBTOOL) --mode=install  $(INSTALL) libmoira.la $(DESTDIR)$(libdir)/libmoira.la
+	$(SRCTOP)/mkinstalldirs $(DESTDIR)$(includedir)/moira
+	$(INSTALL) mr_et.h krb_et.h ureg_err.h $(DESTDIR)$(includedir)/moira
 
 libmoira.la: $(OBJS)
 	$(LIBTOOL) --mode=link $(CC) $(ALL_LDFLAGS) -o $@ $(OBJS) -rpath $(libdir) -version-info 0:0:0
Index: update/Makefile.in
===================================================================
--- update/Makefile.in	(revision 3945)
+++ update/Makefile.in	(working copy)
@@ -4,7 +4,7 @@
 @SET_MAKE@
 
 CC=@CC@
-CPPFLAGS=@CPPFLAGS@
+CPPFLAGS=@CPPFLAGS@ -DCONFIG_FILE=\"$(sysconfdir)/athena/moira.conf\"
 CFLAGS=@CFLAGS@
 DEFS=@DEFS@
 ALL_CFLAGS=$(CPPFLAGS) $(CFLAGS) $(DEFS)
@@ -50,8 +50,8 @@
 install: all
 	$(LIBTOOL) --mode=install $(INSTALL_PROGRAM) update_test $(DESTDIR)$(bindir)
 	$(LIBTOOL) --mode=install $(INSTALL_PROGRAM) update_server $(DESTDIR)$(sbindir)
-	$(SRCTOP)/mkinstalldirs $(DESTDIR)/etc/athena
-	$(INSTALL) -m 644 moira.conf $(DESTDIR)/etc/athena/moira.conf
+	$(SRCTOP)/mkinstalldirs $(DESTDIR)$(sysconfdir)/athena
+	$(INSTALL) -m 644 moira.conf $(DESTDIR)$(sysconfdir)/athena/moira.conf
 
 update_test: $(COBJS) $(MR_LIBDEP)
 	$(LIBTOOL) --mode=link $(CC) -o $@ $(LDFLAGS) $(COBJS) $(LIBS)
Index: update/config.c
===================================================================
--- update/config.c	(revision 3945)
+++ update/config.c	(working copy)
@@ -26,8 +26,6 @@
 
 RCSID("$Header: /afs/.athena.mit.edu/astaff/project/moiradev/repository/moira/update/config.c,v 1.7 1998-02-15 17:49:27 danw Exp $");
 
-#define CONFIG_FILE	"/etc/athena/moira.conf"
-
 /* Variables currently supported:
  * chroot directory	daemon will run chrooted to this directory
  * user username	daemon will run with this user's uid
Index: configure.in
===================================================================
--- configure.in	(revision 3945)
+++ configure.in	(working copy)
@@ -461,7 +461,7 @@
 MR_INCLUDE('$(BUILDTOP)/lib')
 MR_INCLUDE('$(BUILDTOP)/db')
 MR_INCLUDE('$(srcdir)')
-LDFLAGS="$LDFLAGS -L\$(BUILDTOP)/lib $LIBPATH"
+LDFLAGS="$LDFLAGS $LIBPATH"
 KLIBS="$KRB4_LIBS $KRB5_LIBS -lcom_err $LIBS"
 KLIBS="$LIBS"
 if test "$hesiod" != no; then
