--- Makefile.in.orig	2007-12-19 22:04:11.000000000 -0500
+++ Makefile.in	2007-12-19 22:12:17.000000000 -0500
@@ -22,6 +22,7 @@
 LIBS=@LIBS@
 ALL_CFLAGS=-I. -DSYSCONFDIR=\"${sysconfdir}\" ${CPPFLAGS} ${CFLAGS}
 OBJS=hesiod.lo hesmailhost.lo hespwnam.lo hesservbyname.lo hescompat.lo
+LIBTOOLFLAGS=$(foreach flag,$(LDFLAGS),-Xlinker $(flag))
 
 .SUFFIXES: .lo
 
@@ -29,7 +30,7 @@
 
 libhesiod.la: ${OBJS}
 	${LIBTOOL} --mode=link ${CC} -rpath ${libdir} -version-info 0:0:0 \
-	  -o $@ ${OBJS} ${LIBS}
+	  ${LIBTOOLFLAGS} -o $@ ${OBJS} ${LIBS}
 
 hestest: hestest.lo libhesiod.la
 	${LIBTOOL} --mode=link ${CC} -o $@ hestest.lo libhesiod.la
--- hesiod.c.orig	2007-12-17 01:54:47.000000000 -0500
+++ hesiod.c	2007-12-17 02:04:47.000000000 -0500
@@ -45,7 +45,7 @@
 
 #include <sys/types.h>
 #include <netinet/in.h>
-#include <arpa/nameser.h>
+#include <arpa/nameser_compat.h>
 #include <errno.h>
 #include <netdb.h>
 #include <resolv.h>
