diff -ur moira.orig/clients/mrtest/mrtest.c moira/clients/mrtest/mrtest.c
--- moira.orig/clients/mrtest/mrtest.c	2008-08-30 18:22:14.000000000 -0400
+++ moira/clients/mrtest/mrtest.c	2009-01-04 03:20:08.000000000 -0500
@@ -97,8 +97,13 @@
 	}
     }
 
+#if defined(__APPLE__) && defined(__MACH__)
+  add_error_table(&et_sms_error_table);
+  add_error_table(&et_krb_error_table);
+#else
   initialize_sms_error_table();
   initialize_krb_error_table();
+#endif
 
 #ifdef HAVE_READLINE
   /* we don't want filename completion */
diff -ur moira.orig/lib/mr_init.c moira/lib/mr_init.c
--- moira.orig/lib/mr_init.c	2008-08-30 18:22:14.000000000 -0400
+++ moira/lib/mr_init.c	2009-01-04 03:19:06.000000000 -0500
@@ -20,8 +20,15 @@
   if (mr_inited)
     return;
 
+#if defined(__APPLE__) && defined(__MACH__)
+  add_error_table(&et_sms_error_table);
+  add_error_table(&et_krb_error_table);
+  add_error_table(&et_ureg_error_table);
+#else
   initialize_sms_error_table();
   initialize_krb_error_table();
   initialize_ureg_error_table();
+#endif
+
   mr_inited = 1;
 }
