--- copyright.orig	2008-01-27 13:59:15.000000000 -0500
+++ copyright	2008-01-27 13:59:10.000000000 -0500
@@ -0,0 +1,20 @@
+The source code for this package was obtained from
+ftp://athena-dist.mit.edu/pub/ATHENA/hesiod, and is licensed as follows:
+
+  Copyright 1988, 1996 by the Massachusetts Institute of Technology.
+  
+  Permission to use, copy, modify, and distribute this
+  software and its documentation for any purpose and without
+  fee is hereby granted, provided that the above copyright
+  notice appear in all copies and that both that copyright
+  notice and this permission notice appear in supporting
+  documentation, and that the name of M.I.T. not be used in
+  advertising or publicity pertaining to distribution of the
+  software without specific, written prior permission.
+  M.I.T. makes no representations about the suitability of
+  this software for any purpose.  It is provided "as is"
+  without express or implied warranty.
+
+This fink package was created as part of the MacAthena Project 
+<http://macathena.mit.edu/> of the MIT Student Information Processing
+Board, and is released to the public domain.
